### ISSUE TITLE
Data files installed without read permission with Astropy 1.3.1

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#5912]
@lpsinger filed this issue in the Macports tracker:
https://trac.macports.org/ticket/53865

Since Astropy 1.3.1 the "data files" from Astropy are installed in Macports without file permission that prevent them being read by non-root users and thus `import astropy` fails because `astropy.cfg` in site-packages can't be read.

This is a bit surprising since I think the issue was introduced with the Astropy 1.3.0 -> 1.3.1 update, with no changes on the Macports side:
https://github.com/macports/macports-ports/commit/5c14ebc3d35e816cebb24ce08f85b1deec473eb6#diff-9bf981826bafa297ea70fa2ff3e20a21

We had noticed a while back a similar file permission issue for `photutils` in Macports, but didn't manage to track down the cause (and there it only affected running the tests, not photutils itself):
https://github.com/macports/macports-ports/pull/214
If someone knows the Astropy build system well, looking at this full log (for the photutils build) might reveal the issue:
https://gist.github.com/cdeil/88f8e48d85d1c65f593d81db1e7e0d92#file-gistfile1-txt-L1151

@astrofrog @bsipocz - Has there been any change on the Astropy side, e.g. in astropy-helpers, that could lead to installing files without the "read" premission flag set?

@lpsinger @hamogu as Macports users -- any idea what the cause or fix here could be?

I'm pretty clueless how to debug or fix this issue. So unless someone here comes with a suggestion within the next day or two, I'll roll back the update to the previous 1.3.0 in Macports.

**Comment by saimn** [comment]
It seems that there is an issue with the permissions in the tar.gz archive for 1.3.1.
If I download from PyPI:

```
❯ ls -lh astropy-1.3
total 340K
drwxr-xr-x 22 simon 4,0K 23 déc.  07:08 astropy/
drwxr-xr-x  5 simon 4,0K 23 déc.  07:08 astropy_helpers/
drwxr-xr-x  6 simon 4,0K 23 déc.  07:08 cextern/
drwxr-xr-x 23 simon 4,0K 23 déc.  07:08 docs/
drwxr-xr-x  2 simon 4,0K 23 déc.  07:08 licenses/
drwxr-xr-x  2 simon 4,0K 23 déc.  07:08 static/
-rw-r--r--  1 simon  36K 23 déc.  01:28 ah_bootstrap.py
-rw-r--r--  1 simon 236K 23 déc.  06:38 CHANGES.rst
-rw-r--r--  1 simon  13K 23 déc.  01:28 ez_setup.py
-rw-r--r--  1 simon   13 23 déc.  01:28 pip-requirements
-rw-r--r--  1 simon  174 23 déc.  01:28 pip-requirements-dev
-rw-r--r--  1 simon   67 23 déc.  01:28 pip-requirements-doc
-rw-r--r--  1 simon 1,3K 23 déc.  07:08 PKG-INFO
-rw-r--r--  1 simon 1,7K 23 déc.  01:28 README.rst
-rw-r--r--  1 simon 1,3K 23 déc.  01:28 setup.cfg
-rwxr-xr-x  1 simon 3,9K 23 déc.  06:38 setup.py*

❯ ls -lh astropy-1.3.1
total 352K
drwxr-x--- 22 simon 4,0K 18 mars  04:58 astropy/
drwxr-x---  5 simon 4,0K 18 mars  04:58 astropy_helpers/
drwxr-x---  6 simon 4,0K 18 mars  04:58 cextern/
drwxr-x--- 23 simon 4,0K 18 mars  04:58 docs/
drwxr-x---  5 simon 4,0K 18 mars  04:58 examples/
drwxr-x---  2 simon 4,0K 18 mars  04:58 licenses/
drwxr-x---  2 simon 4,0K 18 mars  04:58 static/
-rw-r-----  1 simon  36K 18 mars  02:50 ah_bootstrap.py
-rw-r-----  1 simon 241K 18 mars  04:27 CHANGES.rst
-rw-r-----  1 simon  13K  5 mars  21:30 ez_setup.py
-rw-r-----  1 simon   13  5 mars  21:30 pip-requirements
-rw-r-----  1 simon  174  5 mars  21:30 pip-requirements-dev
-rw-r-----  1 simon   67  5 mars  21:30 pip-requirements-doc
-rw-r-----  1 simon 1,2K 18 mars  04:58 PKG-INFO
-rw-r-----  1 simon 1,7K 18 mars  04:17 README.rst
-rw-r-----  1 simon 1,7K 17 mars  19:52 setup.cfg
-rwxr-x---  1 simon 3,7K 18 mars  04:29 setup.py*
```

**Comment by cdeil** [comment]
@saimn - Thanks for figuring out the origin of the issue.

Indeed, like @saimn said, the permissions in the Astropy 1.3.1 release by @bsipocz are missing the "read" permission flag for all users:
```
-rw-r-----  0 bsipocz staff    1711 Mar 17 19:52 astropy-1.3.1/setup.cfg
```
The same is the case for the photutils 0.3.1 release by @larrybradley where I first noticed the issue:
```
-rw-------  0 lbradley STSCI\science    534 Mar  2 16:36 photutils-0.3.1/setup.cfg
```

I tried making an sdist from the Astropy v1.3.1 tag and file permissions look ok:
```
-rw-r--r--  0 deil   staff    1711 Mar 17 20:40 astropy-1.3.1/setup.cfg
```

First of all - do you agree that the permissions are wrong for the "data files" in the release tarballs?
@bsipocz @larrybradley - can you try to identify why this is happening on your machines?

**Comment by astrofrog** [comment]
Things to check: operating system, python version, setuptools version

**Comment by astrofrog** [comment]
Just to add a datapoint, things look ok for me:

```
-rw-r--r--   1 tom  staff    1711 Mar 27 10:01 setup.cfg
```

MacOS X 10.11
Using Miniconda
Python 3.6.0 :: Continuum Analytics, Inc.
Setuptools: 34.2.0

**Comment by bsipocz** [comment]
@cdeil - ouch, indeed I have a default 0027 umask. 
I would have never though that a release could inherits local permission settings, this definitely needs to go into the release instructions.

@astrofrog @eteq - What should we do here? A 1.3.2 release asap, or somehow release 1.3.1?

**Comment by astrofrog** [comment]
We can't re-release 1.3.1 (PyPI doesn't allow overwrites). I *think* the correct thing would be to release 1.3.1.post1: see https://www.python.org/dev/peps/pep-0440/#post-releases - @eteq what do you think?

**Comment by astrofrog** [comment]
I think the right way to do a post release might be to simply check out the v1.3.1 tag, then change ``__version__`` to 1.3.1.post1 temporarily and do the release, since I don't think anything needs to change in the repo for this fix. Maybe @eteq can confirm though.

**Comment by lpsinger** [comment]
It also says:

> Creating post-releases of pre-releases is strongly discouraged, as it makes the version identifier difficult to parse for human readers. In general, it is substantially clearer to simply create a new pre-release by incrementing the numeric component.

**Comment by MSeifert04** [comment]
Yes, I don't think additional tags make sense for official releases, when I would access `astropy.__version__` and it shows `1.3.1.post1` I would think I got an development version by accident. For `rc` or `dev` it makes sense but these have to be explicitly requested when `pip`-installed.

**Comment by saimn** [comment]
There was a `post1` in the past, with some quirks:  #4406 (in particular https://github.com/astropy/astropy/issues/4406#issuecomment-166372262, if I understand correctly `__version` would still be 1.3.1 with this method). But 1.3.2 is probably simpler/safer.

**Comment by bsipocz** [comment]
@lpsinger - this wasn't a pre-release, so in theory it's fine to make a post release. 
If we go with 1.3.2, I would wait for #5887 to be merged, and make it a real bugfix release rather than just a pseudo that fixes the permissions.

**Comment by larrybradley** [comment]
@cdeil Indeed the `photutils` permissions are also because of my default umask.

This issue must affect only MacPorts installations because the same limited permissions are in *every* `photutils` release, while https://github.com/macports/macports-ports/pull/214 was the first mention of any permission issues.

**Comment by cdeil** [comment]
> This issue must affect only MacPorts installations because the same limited permissions are in every photutils release, while macports/macports-ports#214 was the first mention of any permission issues.

@larrybradley - https://github.com/macports/macports-ports/pull/214 was the initial packaging of photutils in Macports. The same issue might be in earlier releases or not, I didn't check.

My understanding is that this issue is not Macports specific. Won't it happen to anyone that does install with `sudo` (e.g. with pip) and then tries to use the package as non-root user?

**Comment by bsipocz** [comment]
@cdeil @eteq - Should I add a changelog entry for fixing the permissions in 1.3.2 under ``Other Changes``?

**Comment by cdeil** [comment]
+1 to mention it in the changelog. No opinion where exactly.

Do we know which packages besides Astropy / photutils are affected?

Is it possible to automate this check? Do we have a script that does checks on a release sdist before posting where a check on file permissions could be added?

**Comment by bsipocz** [comment]
Nope, we don't yet have such a script. It will affect all packages where the releaser has protective umask, as checking the permissions wasn't yet mentioned in any of the release notes.

**Comment by bsipocz** [comment]
@cdeil - v1.3.2 is now available on PyPI and the permissions seems to be OK. Please reopen if you still see issues.

**Comment by cdeil** [comment]
@bsipocz - Thanks for cutting a new release.

Fix for Macports users will come via:
https://github.com/macports/macports-ports/pull/402

As commented there, indeed now everything looks fine, the file permission issue is resolved with 1.3.2.

**Comment by cdeil** [comment]
For any Macports users, this issue should now be fixed. To update:
```
sudo port selfupdate
sudo port -p upgrade outdated
```
and to check the version number
```
port installed *astropy*
```

**Comment by larrybradley** [comment]
@cdeil, I just released `photutils 0.3.2` to fix the permissions issue.

**Comment by cdeil** [comment]
@larrybradley - Thanks!

I can confirm that the permissions issue I saw with macports earlier is resolved with the photutils 0.3.2 release. Update in Macports will come in https://github.com/macports/macports-ports/pull/405 .

**Comment by embray** [comment]
As @saimn pointed out, if in the future there's a problem just with the release files you can make a `.postN` release.  This should generally work fine now.
